packages/adapter-squid/src/SquidAdapter.ts

```diff
@@ -0,0 +1,226 @@
+import {
+   ConfigProvider,
+   ERR_ADAPTER_CONFIG_MISSING,
+   ERR_ADAPTER_EXECUTE_FAILURE,
+   ERR_ADAPTER_INIT_FAILURE,
+   ERR_GLOBAL_VARIABLE_MISSING,
+   ExecutionResult,
+   IAdapter,
+   IConfigProvider,
+   IValidator,
+   Step,
+   ValidationError,
+   ValidationResult,
+} from "@mate/sdk";
+import {
+   BridgeSwapConfigCodec,
+   SquidAdapterConfig,
+   SquidStepConfig,
+} from "./types";
+import { inject, injectable } from "tsyringe";
+import {
+   ISignerService,
+   ISquidProvider,
+   SignerService,
+   SquidProvider,
+} from "./services";
+import { AdapterConfigValidator, StepConfigValidator } from "./validation";
+import { GetRoute, RouteData, Squid } from "@0xsquid/sdk";
+import { ethers } from "ethers";
+import { TypeOf } from "io-ts";
+import {
+   ERR_GET_ROUTE_FAILURE,
+   ERR_TX_RECEIPT_MISSING,
+   ERR_GET_ROUTE_UNDEFINED,
+   UNKNOWN_SQUID_ERROR,
+} from "./constants";
+import axios from "axios";
+
+@injectable()
+export class SquidAdapter
+   implements IAdapter<ExecutionResult, SquidStepConfig>
+{
+   public adapterId: string = "squid";
+
+   private adapterConfig: SquidAdapterConfig;
+
+   constructor(
+      @inject(AdapterConfigValidator)
+      private adapterConfigValidator: IValidator<SquidAdapterConfig>,
+      @inject(StepConfigValidator)
+      private stepConfigValidator: IValidator<SquidStepConfig>,
+      @inject(SquidProvider) private squidProvider: ISquidProvider,
+      @inject(ConfigProvider) private configProvider: IConfigProvider,
+      @inject(SignerService) private signerService: ISignerService
+   ) {}
+
+   public async init(): Promise<Boolean> {
+      if (this.adapterConfig) {
+         return true;
+      }
+
+      const adapterConfigItem = this.configProvider.getAdapterConfig(
+         this.adapterId
+      );
+
+      if (!adapterConfigItem) {
+         throw new Error(
+            `${ERR_ADAPTER_INIT_FAILURE}: ${ERR_ADAPTER_CONFIG_MISSING}`
+         );
+      }
+
+      try {
+         this.adapterConfig = await this.adapterConfigValidator.validate(
+            adapterConfigItem.config
+         );
+
+         const squidConfig = {
+            baseUrl: this.adapterConfig.baseUrl,
+            integratorId: this.adapterConfig.integratorId,
+         };
+
+         await this.squidProvider.init(squidConfig);
+      } catch (error: any) {
```

Comment:
nit: slightly safer to use `unknown` in catch clauses AFAIK to avoid unexpected error types (or at least deal with them explicitly). There's a compiler option that makes `unknown` the default in catch clauses https://www.typescriptlang.org/tsconfig#useUnknownInCatchVariables


```suggestion
      } catch (error: unknown) {
```

Reply:
This is good. Will update everywhere 👌